src/primitives/felt.ts

```diff
@@ -1,61 +1,54 @@
 import { MaybeRelocatable, Relocatable } from './relocatable';
 
-import {
-  ForbiddenOperation,
-  OutOfRangeBigInt,
-  PrimitiveError,
-} from 'errors/primitives';
+import { ForbiddenOperation, OutOfRangeBigInt } from 'errors/primitives';
 
 export class Felt {
   private inner: bigint;
   static PRIME: bigint =
     0x800000000000011000000000000000000000000000000000000000000000001n;
   static ZERO: Felt = new Felt(0n);
   constructor(_inner: bigint) {
-    if (_inner < 0n || _inner > Felt.PRIME) {
-      throw new PrimitiveError(OutOfRangeBigInt);
-    }
-    this.inner = _inner;
+    while (_inner < 0n) _inner += Felt.PRIME;
```

Comment:
why not just `this.inner = _inner % Felt.PRIME` ?

Reply:
The `inner` attribute should be positive-only, but Javascript modulo returns the sign of the dividend
(`-9 % 5 = 1`) (wanted behaviour)
(`-9 % 5 = -4`) (JavaScript behaviour)

To have postive-only modulo with `%` there is this [trick](https://web.archive.org/web/20090717035140if_/javascript.about.com/od/problemsolving/a/modulobug.htm):
`this.inner = (_inner % Felt.PRIME) + Felt.PRIME) % Felt.PRIME)`

But I feel like it reduces readability

One suggestion could be

```suggestion
    if (_inner < 0n) {
      _inner = _inner % Felt.PRIME + Felt.PRIME
    }
```